src/contracts/libraries/BeaconChainProofs.sol

```diff
@@ -279,11 +282,15 @@ library BeaconChainProofs {
             "BeaconChainProofs.verifyWithdrawalProofs: timestampProof has incorrect length");
 
         if(proofs.proveHistoricalRoot){
+            //calculate the blockHeaderRoot Index for a block that is very old
             uint256 historicalBlockHeaderIndex = HISTORICAL_SUMMARIES_INDEX << ((HISTORICAL_SUMMARIES_TREE_HEIGHT + 1) + 1 + (BLOCK_ROOTS_TREE_HEIGHT)) | 
-                                                BLOCK_SUMMARY_ROOT_INDEX << (BLOCK_ROOTS_TREE_HEIGHT) | uint256(proofs.blockHeaderRootIndex);
-        }
+                                                uint256(proofs.historicalSummaryIndex) << (1 + (BLOCK_ROOTS_TREE_HEIGHT)) |
+                                                BLOCK_SUMMARY_ROOT_INDEX << (BLOCK_ROOTS_TREE_HEIGHT) | 
+                                                uint256(proofs.blockHeaderRootIndex);
 
-        {
+            require(Merkle.verifyInclusionSha256(proofs.historicalSummaryBlockRootProof, beaconStateRoot, proofs.blockHeaderRoot, historicalBlockHeaderIndex),
+                "BeaconChainProofs.verifyWithdrawalProofs: invalid historical summary proof");
+        } else {
```

Comment:
what happens if I mark `proofs.proveHistoricalRoot` as 'false' but I supply a really old blockHeaderRoot?
where should the process fail?
do we have test coverage for this?

Reply:
doesnt exist, all the historical proof stuff is a wip

Reply:
I added a TODO + note in the code about this for now, at least.